File: datasets/IR-Plag/convertai-original-java-rest-python/case-07/plagiarized/L1/04/multiarray.py
import sys

def sumMajorDiagonal(m):
    """
    Calculates the sum of the elements in the major diagonal of a 2D matrix.
    
    Args:
        m (list of lists): A 2D matrix represented as a list of lists.
    
    Returns:
        float: The sum of the elements in the major diagonal.
    """
    sum = 0
    for i in range(len(m)):
        sum += m[i][i]
    return sum

if __name__ == "__main__":
    print("Enter a 4 by 4 matrix row by row: ", end="")
    m = []
    for i in range(4):
        row = [float(x) for x in input().split()]
        m.append(row)
    
    print(f"Sum of the elements in the major diagonal is {sumMajorDiagonal(m)}")

File: datasets/IR-Plag/convertai-original-java-rest-typescript/case-07/plagiarized/L1/05/multiarray.ts
import * as readline from 'readline';

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

function sumMajorDiagonal(m: number[][]): number {
  let sum = 0;
  for (let i = 0; i < m.length; i++) {
    sum += m[i][i];
  }
  return sum;
}

rl.question('Enter a 4 by 4 matrix row by row: ', (input) => {
  const m: number[][] = [];
  const rows = input.trim().split('\n');
  for (let i = 0; i < 4; i++) {
    const row = rows[i].trim().split(' ').map(Number);
    m.push(row);
  }

  console.log(`Sum of the elements in the major diagonal is ${sumMajorDiagonal(m)}`);
  rl.close();
});

